src/game_board.cpp

```diff
@@ -182,6 +182,24 @@ void game_board::check_victory(bool& continue_level,
 	continue_level = false;
 }
 
+void game_board::set_affect_distant_max_radius(std::optional<int> value)
+{
+	if(value){
+		if(!affect_distant_max_radius_ || *affect_distant_max_radius_ < *value){
+			affect_distant_max_radius_ = value;
+		}
+	}
+}
+
+void game_board::set_affect_distant_max_radius(const config& cfg)
+{
+	for(const config& affect_distant : cfg.child_range("affect_distant")) {
+		if(affect_distant.has_attribute("radius")){
+			set_affect_distant_max_radius(affect_distant["radius"].to_int());
+		}
+	}
+}
```

Comment:
Storing this on the gameboard seems odd, since it's related to a particular unit's ability.

Reply:
Except that the check is always centered on the unit involved in any action, first if it has an ability affecting itself, then check if an adjacent unit has the ability with [affect_adjacent] and matches the target unit. Because of this, I can't create a max_radius variable in the unit class because if the target unit doesn't have an ability with [affect_distant], max_radius will always be zero.

That's why I have to use a scenario-specific variable like the game_board or game_state class for this to work. A single max_radius corresponds to the highest [affect_distant]radius= value. If no unit on the board or in the recall list has one, the check won't take place.

Reply:
i tried with abilities class without what it's work

Reply:
So essentially it needs to check whether any unit within that radius has an ability that affects the current unit that's doing an action?

Reply:
Yes, its why max_radius is equals to radius of ability with affect_distant with greater radius

Reply:
Is that fully reliable? Given how abilities can affect other abilities, it seems like you'd need to check every unit on the map. As an example, say you have:
* Unit A at 1,1
* Unit B at 1,5 with an ability with a radius of 5
* Unit C at 1,12 with an ability with a radius of 7

If Unit B's ability being active is dependent on Unit C's ability, would that be handled correctly?

Reply:
Do you know when it starts to become a performance problem? Does it cause any problems to have 20 units with [affect_distant] that all affect each other, for example?

Reply:
Wih map with 4 or 5 sides, it go with more of 20 units, and same if i don't max_radius variables, i should replace that by a boolean variable for prevent check of all unit if none own abilities with affect_distant and with max_radius, o part of units only is checked each time

Reply:
Right, so it'd be important to know at what point it starts to become a performance problem on a build with optimizations enabled. If nothing else it would be something to note on the wiki, but also if it starts having performance problems with a small number of units then that would raise the question of how useful it could really be in practice.

Reply:
optimizations enabled: this is the use of game_board variables to regulate unit checks, isn't it?